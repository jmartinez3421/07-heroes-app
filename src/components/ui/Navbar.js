import React, { useContext } from 'react'
import { Link, NavLink, useHistory } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext'
import { types } from '../../types/types';

export const Navbar = () => {

    const {user, dispatch} = useContext(AuthContext);
    const history = useHistory();

    const handleLogout = () => {
        
        history.replace('/07-heroes-app/login');
        dispatch({
            type: types.logout
        })
    }

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark ps-2 pe-2">
            
            <Link 
                className="navbar-brand" 
                to="/07-heroes-app"
            >
                Heroes App
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/07-heroes-app/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/07-heroes-app/dc"
                    >
                        DC
                    </NavLink>

                    <NavLink 
                        activeClassName='active'
                        className='nav-item nav-link'
                        exact
                        to='/07-heroes-app/search'
                    >
                        Search
                    </NavLink>
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
                <ul className="navbar-nav ms-auto">
                    <span className='nav-item nav-link text-info' id='username'>
                        {user.name}
                    </span>
                    <button 
                        className="nav-item nav-link btn" 
                        onClick={handleLogout}
                    >
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    )
}