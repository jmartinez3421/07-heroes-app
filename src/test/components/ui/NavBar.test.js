import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter, Router } from 'react-router';
import { AuthContext } from '../../../auth/AuthContext';
import { Navbar } from '../../../components/ui/Navbar';
import { types } from '../../../types/types';
import { contextValue } from '../../fixtures/contextValue';


describe('Pruebas sobre el componente <NavBar />', () => {

    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn()
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>    
    );

        afterEach(() => {
            jest.clearAllMocks();
        })

    test('Debe mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('span#username').text().trim()).toBe(contextValue.user.name);
    });
    
    test('Debe de llamar el logout y usar history al pulsar el botÃ³n', () => {
        wrapper.find('button').prop('onClick')();

        expect(contextValue.dispatch).toHaveBeenCalledWith({type: types.logout});
        expect(historyMock.replace).toHaveBeenCalledWith('/login');
    })
    
})
