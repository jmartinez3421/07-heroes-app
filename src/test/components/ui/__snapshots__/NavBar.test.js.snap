// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <NavBar /> Debe mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark ps-2 pe-2"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Heroes App
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ms-auto"
    >
      <span
        className="nav-item nav-link text-info"
        id="username"
      >
        Jordi
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
